.golangci.yml

```diff
@@ -0,0 +1,283 @@
+    - format
+    # depguard, dupl, gochecknoglobals, gochecknoinits, goconst, gocritic,
+    # golint, gosimple, interfacer, lll, misspell, stylecheck, unconvert
+    - style
+  fast: false
+
+
+issues:
+  # List of regexps of issue texts to exclude, empty list by default.
+  # But independently from this option we use default exclude patterns,
+  # it can be disabled by `exclude-use-default: false`. To list all
+  # excluded by default patterns execute `golangci-lint run --help`
+  exclude:
+    # Exclude table-driven tests from scopelint (https://github.com/golangci/golangci-lint/issues/281).
+    - "Using the variable on range scope `tt` in function literal"
+    - "Using the variable on range scope `test` in function literal"
+    # It's common to shadow `err` and rarely indicates a problems. See
+    # https://github.com/golang/go/issues/19490 for further details.
+    - 'shadow: declaration of "err" shadows declaration'
+    # We commonly expose profiling information on /debug/pprof so we need to disable the gosec
+    # lint for it.
+    - "Profiling endpoint is automatically exposed on /debug/pprof"
+    # The logger is often our last option to communicate that an error occurred so if it returns
+    # an error we don't have an alternative to use. Since it's already unlikely that `Log` will
+    # return an error anyway we often skip checking the error for brevity.
+    - "Error return value of `\\(github.com\\/go-kit\\/kit\\/log.Logger\\).Log` is not checked"
+    # The caller is responsible for closing the Body of an `http.Response`. However, this step
+    # is usually performed in a defer function after the response has already been processed and
+    # so errors, which are already rare, can usually be safely ignored.
+    - "Error return value of `[a-zA-Z.]+.Body.Close` is not checked"
+    # The errcheck linter already checks for unhandled errors so we can disable the equivalent
+    # lint by gosec.
+    - "G104: Errors unhandled"
+
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    # Exclude lll issues for long lines with go:generate
+    - linters:
+        - lll
+      source: "^//go:generate "
+    # Exclude some linters from running on tests files.
+    # - path: _test\.go
+    #   linters:
+    #     - gocyclo
+    #     - errcheck
+    #     - dupl
+    #     - gosec
+
+
+  # Independently from option `exclude` we use default exclude patterns,
+  # it can be disabled by this option. To list all
+  # excluded by default patterns execute `golangci-lint run --help`.
+  # Default value for this option is true.
+  exclude-use-default: false
+
+  # Maximum issues count per one linter. Set to 0 to disable. Default is 50.
+  max-issues-per-linter: 0
+
+  # Maximum count of issues with the same text. Set to 0 to disable. Default is 3.
+  max-same-issues: 0
+
+  # Show only new issues created after git revision `REV`
+  new-from-rev: a0ecbf9d10486673d7581f4e6a8b0f30497f8a66
```

Comment:
How many errors are there if we remove this? I.e. is it worth it to bite the bullet now and fix them?

Reply:
Never mind, it looks pretty gnarly. This SGTM.

Reply:
Around 600 errors. Top 5 linters by error count:

```
 138 paralleltest
 115 gci
 105 lll
  44 errcheck
  30 errorlint
```